Lib/test/test_concurrent_futures.py

```diff
@@ -1235,12 +1235,5 @@ def test_multiple_set_exception(self):
         self.assertEqual(f.exception(), e)
 
 
-@test.support.reap_threads
-def test_main():
-    try:
-        test.support.run_unittest(__name__)
-    finally:
-        test.support.reap_children()
-
 if __name__ == "__main__":
-    test_main()
+    unittest.main()
```

Comment:
LGTM :-)

Reply:
Oh wait. You removed @test.support.reap_threads. Maybe it would help to add setUpModule() / tearDownModule() to call threading_setup() / threading_cleanup() (warning: the API is little bit special, search for examples in other tests).

You may also call test.support.reap_children() in tearDownModule().